### PR TITLE
Action Cable: client side 💅

### PR DESCRIPTION
**Pull request by javan**
- Moves the default mount path to `ActionCable::INTERNAL` to ensure it stays synced with `action_cable.js` (via `action_cable.coffee.erb`)
- Restores `ActionCable#getConfig` to its original, intended design for reading _any_ `action-cable-*` meta tag
- Minor internal refactoring for clarity and convenience

/cc @maclover7, @jeremy

### CODE REVIEW

File: actioncable/app/assets/javascripts/action_cable/subscription.coffee [hunk 1]
```diff
@@ -44,11 +44,9 @@
 # The "AppearanceChannel" name is automatically mapped between the client-side subscription creation and the server-side Ruby class name.
 # The AppearanceChannel#appear/away public methods are exposed automatically to client-side invocation through the @perform method.
 class ActionCable.Subscription
-  constructor: (@subscriptions, params = {}, mixin) ->
+  constructor: (@consumer, params = {}, mixin) ->
     @identifier = JSON.stringify(params)
     extend(this, mixin)
-    @subscriptions.add(this)
```
Comment (maclover7) [inline actioncable/app/assets/javascripts/action_cable/subscription.coffee]: Where is this logic performed elsewhere, I think we still want to do this? Could be missing something -- not super familiar with CoffeeScript :)

**Comment by maclover7** [comment]
In general, :+1: for refactoring this a bit.

File: actioncable/app/assets/javascripts/action_cable/subscription.coffee (hunk 1 above)
Comment (javan) [inline actioncable/app/assets/javascripts/action_cable/subscription.coffee]: Moved it to `Subscriptions#create` (see ~13 lines down in this PR). Before, `Subscriptions#create` constructed `Subscription` instances with a reference to `@subscriptions` and the subscription would then add itself to the collection. I'm not sure why I implemented that way originally. Now, `Subscriptions#create` [makes a `Subscription` and then adds it to its collection](https://github.com/javan/rails/blob/actioncable-cleanup/actioncable/app/assets/javascripts/action_cable/subscriptions.coffee#L16-L17). 

In short: same end result, less confusing circular object passing.

Comment (maclover7) [inline actioncable/app/assets/javascripts/action_cable/subscription.coffee]: Ahh, gotcha. :+1: for less circular logic. :)

**Comment by jeremy** [comment]
❤️
